flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/AggregateUtil.scala

```diff
@@ -424,16 +430,27 @@ object AggregateUtil extends Enumeration {
       hasStateBackedDataViews: Boolean)
     : AggregateInfo = {
 
-    // TODO handle data views here
+    // extract data views and adapt the data views in the accumulator type
+    // if a view is backed by a state backend
+    val dataViewSpecs: Array[DataViewSpec] = if (hasStateBackedDataViews) {
+      DataViewUtils.extractDataViews(index, accumulatorDataType).toList.toArray
```

Comment:
nit: I think a preferred way is to do: `
```suggestion
      DataViewUtils.extractDataViews(index, accumulatorDataType).asScala.toArray
```

The `toList` does a conversion, whereas `asScala` creates only a wrapper.